app/models/set_descriptors_task.rb

```diff
@@ -1,10 +1,67 @@
-class SetDescriptorsTask < Task # rubocop:todo Style/Documentation
+# The SetDescriptorsTask presents a series of configured {Descriptor descriptors}
+# to the user, and lets them specify values for them. These values are the record
+# in {LabEvent lab events} created for each {Request}.
+#
+# The per_item attribute, specifies whether the user is presented with a separate
+# set of descriptors for each request, or if they are shared across the {Batch}.
+# In either case, the user can deselect individual requests from a table at the top
+# of the screen, to only apply their attributes to a subset of requests.
+class SetDescriptorsTask < Task
+  def partial
+    'set_descriptors'
+  end
+
+  def can_process?(batch, from_previous: false)
+    batch.released? ? [true, 'Edit'] : [true, nil]
+  end
+
   def render_task(workflows_controller, params)
-    super
-    workflows_controller.render_set_descriptors_task(self, params)
+    Tasks::SetDescriptorsHandler::Handler.new(controller: workflows_controller, params: params, task: self).render
```

Comment:
This pretty much ends up still calling what we had before. But gives us a platform to improve upon.